### ISSUE TITLE
Checkstyle formatter does not include error free files.

### ISSUE THREAD

**Issue by mastermatt** [palantir/tslint#3837]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.8.1
- __Running TSLint via__: CLI 

#### TypeScript code being linted

```ts
// error free code
```

with `tslint.json` configuration:

```json
{
    "extends": [
        "tslint:recommended"
    ]
}
```
`tslint --project . --format checkstyle index.ts`

#### Actual behavior (newlines added for readability)
```xml
<?xml version="1.0" encoding="utf-8"?>
<checkstyle version="4.3">
</checkstyle>
```
#### Expected behavior
```xml
<?xml version="1.0" encoding="utf-8"?>
<checkstyle version="4.3">
<file name="/path/to/src/index.ts"></file>
</checkstyle>
```

As noted in https://github.com/palantir/tslint/issues/250, [this file](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/XMLLogger.java) is the closest thing to an official spec on the format. 
The relevant bit is that [`writeFileMessages`](https://github.com/checkstyle/checkstyle/blob/c2269fee863b218e5be4bf77f0f7ec29286d2f2b/src/main/java/com/puppycrawl/tools/checkstyle/XMLLogger.java#L155) writes a tag for each file linted. Errorless files are just empty tags. 
As I'm finding out, some parsers need those tags. Otherwise they return an error message stating no lint checks were preformed and/or they allows show 100% of files have issues.  

I opted to open an issue on this instead of submitting a PR as I'm unsure of the preferred way to handle this. The `Linter` class doesn't store all the files it lints and therefore can't pass them to the formatter in [`getResults()`]() as is. One option would be to just add each file in turn in [`lint()`](https://github.com/palantir/tslint/blob/master/src/linter.ts#L114), then add a third arg to `IFormatter.format`.

**Comment by mastermatt** [comment]
Seems related to https://github.com/palantir/tslint/issues/3744

**Comment by mastermatt** [comment]
This was resolved with https://github.com/palantir/tslint/pull/3838
